portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,32 +14,66 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
-import com.google.gson.Gson;
-import java.util.List;
-import java.util.Arrays;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-
 public class DataServlet extends HttpServlet {
-  
-  private List<String> groceryList;
-
-  public DataServlet(){  
-    groceryList = Arrays.asList("Fruit", "Soap", "Cereal");
-  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(groceryList);
-    response.setContentType("application/json;");
+    final Query commentQuery = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
Like Tong mentioned earlier, we should turn "Comments", "comments", and "timestamp" into variables. That way whenever we need to change them we can change them in one place. And you don't have to worry about typos after you get it right the first time.

Example:
```
public class DataServlet extends HttpServlet {
  final String COMMENT_TABLE = "Comments";
  final String COMMENT_COL = "comments";
  final String TIMESTAMP_COL = "timestamp";
  ...
}
```

And then in your code you might do:

```
entity.getProperty(COMMENT_COL);
```
or
```
commentEntity.setProperty(COMMENT_COL, comments);
```

See: https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names